.github/workflows/cmake.yml

```diff
@@ -25,6 +25,15 @@ jobs:
     runs-on: ${{ matrix.os }}
     container: ${{ matrix.container }}
     steps:
+      - name: Upgrade GCC
+        run: |
+          sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+          sudo apt install -y gcc-11 g++-11
+          sudo update-alternatives \
+            --install /usr/bin/gcc gcc /usr/bin/gcc-11 11 \
+            --slave /usr/bin/g++ g++ /usr/bin/g++-11
+          sudo update-alternatives --set gcc /usr/bin/gcc-11
+        if: matrix.name == 'Linux'
```

Comment:
We should update the docker container too.

Reply:
Yea, that's ideally how the new compiler would get included. This PR upgrades it manually in the meantime.